azurerm/internal/services/mariadb/mariadb_server_resource.go

```diff
@@ -296,10 +295,7 @@ func resourceArmMariaDbServerCreate(d *schema.ResourceData, meta interface{}) er
 	}
 
 	ssl := mariadb.SslEnforcementEnumEnabled
-	if v, ok := d.GetOk("ssl_enforcement"); ok && strings.EqualFold(v.(string), string(mariadb.SslEnforcementEnumDisabled)) {
-		ssl = mariadb.SslEnforcementEnumDisabled
-	}
```

Comment:
since the replacement is a Boolean, and the original field is a Computed String - both fields will always have a value - as such this'd mean that we'd need to continue to configure both fields here, which is why this instead forces the user over to the new field